api/tests/Unit/PoolAuthorizationScopeTest.php

```diff
@@ -0,0 +1,289 @@
+    public function testViewAsPoolOperator(): void
+    {
+        Auth::shouldReceive('user')
+            ->andReturn(User::factory()
+                ->asPoolOperator($this->team1->name)
+                ->create());
+
+        // draft pool from team 1 and three published pools from both teams
+        $poolIds = Pool::authorizedToView()->get()->pluck('id');
+        assertEqualsCanonicalizing([
+            $this->poolDraft1->id,
+            $this->poolPublished1->id,
+            $this->poolClosed1->id,
+            $this->poolArchived1->id,
+            $this->poolPublished2->id,
+            $this->poolClosed2->id,
+            $this->poolArchived2->id,
+        ], $poolIds->toArray());
+    }
+
+    // a request responder should be able to view any published pool (like anyone can)
+    public function testViewAsRequestResponder(): void
+    {
+        Auth::shouldReceive('user')
+            ->andReturn(User::factory()
+                ->asRequestResponder()
+                ->create());
+
+        // three published from both teams
+        $poolIds = Pool::authorizedToView()->get()->pluck('id');
+        assertEqualsCanonicalizing([
+            $this->poolPublished1->id,
+            $this->poolClosed1->id,
+            $this->poolArchived1->id,
+            $this->poolPublished2->id,
+            $this->poolClosed2->id,
+            $this->poolArchived2->id,
+        ], $poolIds->toArray());
+    }
+
+    // a community manager should be able to view all the pools
+    public function testViewAsCommunityManager(): void
+    {
+        Auth::shouldReceive('user')
+            ->andReturn(User::factory()
+                ->asCommunityManager()
+                ->create());
+
+        // four from team 1 and four from team 2
+        $poolIds = Pool::authorizedToView()->get()->pluck('id');
+        assertEqualsCanonicalizing([
+            $this->poolDraft1->id,
+            $this->poolPublished1->id,
+            $this->poolClosed1->id,
+            $this->poolArchived1->id,
+            $this->poolDraft2->id,
+            $this->poolPublished2->id,
+            $this->poolClosed2->id,
+            $this->poolArchived2->id,
+        ], $poolIds->toArray());
+    }
+
+    // a platform admin should be able to view all the pools
+    public function testViewAsPlatformAdmin(): void
+    {
+        Auth::shouldReceive('user')
+            ->andReturn(User::factory()
+                ->asAdmin()
+                ->create());
+
+        // four from team 1 and four from team 2
+        $poolIds = Pool::authorizedToView()->get()->pluck('id');
+        assertEqualsCanonicalizing([
+            $this->poolDraft1->id,
+            $this->poolPublished1->id,
+            $this->poolClosed1->id,
+            $this->poolArchived1->id,
+            $this->poolDraft2->id,
+            $this->poolPublished2->id,
+            $this->poolClosed2->id,
+            $this->poolArchived2->id,
+        ], $poolIds->toArray());
+    }
+}
```

Comment:
Could you add a comment: `TODO: add tests for new roles as part of #10609`

Reply:
[extra comments](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11138/commits/dbda192c713a308da0628b1501ca454a706f879a)